cypress/utils/productsUtils.js

```diff
@@ -0,0 +1,124 @@
+import Product from "../apiRequests/Product";
+
+class ProductsUtils {
+  createdVariantId;
+  productTypeId;
+  attributeId;
+  categoryId;
+
+  updateChannelInProduct(productsList, channelId) {
+    const product = new Product();
+    productsList.forEach(productElement => {
+      product.updateChannelInProduct(productElement.node.id, channelId);
+      const variants = productElement.node.variants;
+      variants.forEach(variant => {
+        product.updateChannelPriceInVariant(variant.id, channelId);
+      });
+    });
+  }
+  createProductInChannel(
+    name,
+    channelId,
+    warehouseId,
+    quantityInWarehouse,
+    productTypeId,
+    attributeId,
+    categoryId,
+    price
+  ) {
+    const product = new Product();
+    return product
+      .createProduct(attributeId, name, productTypeId, categoryId)
+      .then(createProductResp => {
+        const productId = createProductResp.body.data.productCreate.product.id;
+        return product.updateChannelInProduct(productId, channelId).then(() =>
+          product
+            .createVariant(
+              productId,
+              name,
+              warehouseId,
+              quantityInWarehouse,
+              channelId,
+              price
+            )
+            .then(createVariantResp => {
+              this.createdVariantId =
+                createVariantResp.body.data.productVariantBulkCreate.productVariants;
+            })
+        );
+      });
+  }
+
+  createTypeAttributeAndCategoryForProduct(name) {
+    const attribute = new Attribute();
+    const category = new Category();
+    const product = new Product();
+    return attribute.createAttribute(name).then(createAttributeResp => {
+      this.attributeId =
+        createAttributeResp.body.data.attributeCreate.attribute.id;
+      return product
+        .createTypeProduct(name, this.attributeId)
+        .then(createTypeProductResp => {
+          this.productTypeId =
+            createTypeProductResp.body.data.productTypeCreate.productType.id;
+          return category.createCategory(name).then(categoryResp => {
+            this.categoryId = categoryResp.body.data.categoryCreate.category.id;
+          });
+        });
+    });
+  }
+
+  getCreatedVariants() {
+    return this.createdVariantId;
+  }
+  getProductTypeId() {
+    return this.productTypeId;
+  }
+  getAttributeId() {
+    return this.attributeId;
+  }
+  getCategoryId() {
+    return this.categoryId;
+  }
+
+  deleteProducts(startsWith) {
+    const product = new Product();
+    const attribute = new Attribute();
+    const category = new Category();
+    product.getProductTypes(100, startsWith).then(resp => {
+      const productTypes = resp.body.data.productTypes.edges;
+      productTypes.forEach(productType => {
+        if (productType.node.name.includes(startsWith)) {
+          product.deleteProductType(productType.node.id);
+        }
+      });
+    });
+    attribute.getAttributes(100, startsWith).then(resp => {
```

Comment:
These parts are copied pasted. I'd suggest making a class of it, or at least
```suggestion
    const ELEMENTS_AMOUNT = 100

deleteProperCategories = (startsWith) =>  {
    const res => category.getCategories(ELEMENTS_AMOUNT, startsWith)
    const { edges : categories } = res.body.data.categories

    const deleteElement = handleDeleteElement(category.deleteCategory)

    categories.forEach(element => deleteElement(categories, startsWith))
}

handleDeleteElement = (deleteFunction) => (element, startsWith) => {
    if(!isAmongDeletableElements(element, startsWith)) { return; }

    deleteFunction(element.node.id)
}

isAmongDeletableElements = (element, startsWith) => element.node.name.includes(startsWith)
```